diff --git a/src/vs/server/node/extensionHostConnection.ts b/src/vs/server/node/extensionHostConnection.ts
--- a/src/vs/server/node/extensionHostConnection.ts
+++ b/src/vs/server/node/extensionHostConnection.ts
@@ -284,9 +284,9 @@ export class ExtensionHostConnection extends Disposable {
 			const args = ['--type=extensionHost', `--transformURIs`];
 			const useHostProxy = this._environmentService.args['use-host-proxy'];
 			args.push(`--useHostProxy=${useHostProxy ? 'true' : 'false'}`);
-			if (this._configurationService.getValue<boolean>('extensions.supportNodeGlobalNavigator')) {
-				args.push('--supportGlobalNavigator');
-			}
+			// Keep the VSCodium remote extension host compatible with extensions that
+			// still access Node's global navigator during activation.
+			args.push('--supportGlobalNavigator');
 			this._extensionHostProcess = cp.fork(FileAccess.asFileUri('bootstrap-fork').fsPath, args, opts);
 			const pid = this._extensionHostProcess.pid;
 			this._log(`<${pid}> Launched Extension Host Process.`);
