diff --git a/resources/server/bin/code-server-linux.sh b/resources/server/bin/code-server-linux.sh
index 2a0e0e7b475..fb99fd527d6 100755
--- a/resources/server/bin/code-server-linux.sh
+++ b/resources/server/bin/code-server-linux.sh
@@ -16,4 +16,8 @@ if [ -n "$VSCODE_SERVER_CUSTOM_GLIBC_LINKER" ] && [ -n "$VSCODE_SERVER_CUSTOM
 	echo "Patching complete."
 fi
 
+if [ -f "$ROOT/bin/vscodium-preserve-marketplace.js" ]; then
+	"$ROOT/node" "$ROOT/bin/vscodium-preserve-marketplace.js" "$ROOT" || true
+fi
+
 "$ROOT/node" ${INSPECT:-} "$ROOT/out/server-main.js" "$@"
